### PR TITLE
AzureStack - User subscriptions REST API spec

### PR DESCRIPTION
**Pull request by bganapa**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by sarangan12 — APPROVED** [review]

**Review by sarangan12 — CHANGES_REQUESTED** [review]
@bganapa Please resolve the errros reported here: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/365893862

**Comment by bganapa** [comment]
@sarangan12 Thanks for the review. fixed the issues

**Comment by sarangan12** [comment]
Please take a look at https://travis-ci.org/Azure/azure-rest-api-specs/jobs/367409297. There are still some errors to be resolved

**Comment by bganapa** [comment]
@sarangan12 
The only error is
"Parameter \"subscriptionId\" is not allowed in the operations section, define it in the global parameters section instead"

I really want to keep this in the operations section and not in the global parameters. In the context of this spec, this is not a global parameter. This spec is for managing subscriptions in azure stack(like user creating/updating/removing subscription) Azure does not have similar apis.

I believe this justifies the need to keep this NOT as a global parameter.

**Comment by sarangan12** [comment]
@bganapa Look at this: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/367409297#L813 There are 2 more errors...

**Comment by bganapa** [comment]
@sarangan12 . The following error should be ignored for this azurestack subscription spec. Azure stack has an api for creating/updating/deleting subscription resource.

Aligning with azure Subscription is not modelled as a ARM resource in azurestack as well.  You can see here the azure spec where subscription is not an ARM resource

"message": "Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity, zones. Model definition 'Subscription' has extra properties ['displayName,offerId,state,subscriptionId,tenantId'].",
  "i

I believe these rules can be ignored for this spec.

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+                    "Get the specified subscription.": {
+                        "$ref": "./examples/Subscriptions/Get.json"
+                    }
+              },
+              "tags": [
+                "Subscriptions"
+              ],
+              "operationId": "Subscriptions_Get",
+              "description": "Gets details about particular subscription.",
+              "parameters": [
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Why are you not using the SubscriptionIdParameter here?

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+                        "$ref": "./examples/Subscriptions/Create.json"
+                    }
+                },
+                "description": "Create or updates a subscription.",
+                "tags": [
+                    "Subscriptions"
+                ],
+                "operationId": "Subscriptions_CreateOrUpdate",
+                "parameters": [
+                    {
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Why are you not using the SubscriptionIdParameter here?

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+                        "Delete the specifed subscription": {
+                            "$ref": "./examples/Subscriptions/Delete.json"
+                        }
+                    },
+                    "description": "Delete the specifed subscription.",
+                    "tags": [
+                        "Subscriptions"
+                    ],
+                    "operationId": "Subscriptions_Delete",
+                    "parameters": [
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Why are you not using the SubscriptionIdParameter here?

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+            },
+            "LocationParameter": {
+                "description": "The AzureStack location.",
+                "name": "location",
+                "in": "path",
+                "type": "string",
+                "required": true,
+                "x-ms-parameter-location": "method"
+            },
+            "SubscriptionParameter": {
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: I don't see where this is being referenced.

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+            },
+            "ResourceGroupParameter": {
+                "description": "The resource group the resource is located under.",
+                "name": "resourceGroup",
+                "in": "path",
+                "type": "string",
+                "required": true,
+                "x-ms-parameter-location": "method"
+            },
+            "LocationParameter": {
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: I don't see where this is being referenced.

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+            },
+            "ApiVersionParameter": {
+                "name": "api-version",
+                "in": "query",
+                "description": "Client Api Version.",
+                "required": true,
+                "type": "string",
+                "default": "2015-11-01"
+            },
+            "ResourceGroupParameter": {
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: I don't see where this is being referenced.

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json [hunk 7]
```diff
@@ -0,0 +1,412 @@
...
+                        "type": "array",
+                        "items": {
+                            "$ref": "#/definitions/Operation"
+                        }
+                    }
+                }
+            }            
+        },
+        "parameters": {
+            "SubscriptionIdParameter": {
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Not being referenced.  If you don't want it to be global change it to being passed in by a method.

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Offer.json [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+                        "$ref": "#/definitions/Offer"
+                    }
+                },
+                "nextLink": {
+                    "type": "string",
+                    "description": "URI to next page."
+                }
+            }
+        }
+    },
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Offer.json]: You should add parameters even if empty.

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 1 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Subscription is the main resource for this spec, I dont want to mask it in global param

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 2 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Subscription is the main resource for this spec, I dont want to mask it in global param

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 3 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Subscription is the main resource for this spec, I dont want to mask it in global param

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Offer.json (hunk 8 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Offer.json]: Why is that needed? Validators and SDK generation goes fine without it..

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 6 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: removed

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 7 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: removed

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 4 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: removed

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 5 above)
Comment (bganapa) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: removed

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Offer.json (hunk 8 above)
Comment (deathly809) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Offer.json]: If you don't have it then you should remove the empty ones from your other specs.  I think it is best to keep all the high level items there.

File: specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json (hunk 1 above)
Comment (deathly809) [reply specification/azsadmin/resource-manager/user-subscriptions/Microsoft.Subscriptions/preview/2015-11-01/Subscriptions.json]: Then change it from a global parameter to a passed in parameter.  There is no need to repeat the same parameter in multiple places.

File: specification/azsadmin/resource-manager/user-subscriptions/readme.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+## C# 
+
+``` yaml $(csharp)
+csharp:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  namespace: Microsoft.AzureStack.Management.Subscriptions
+  payload-flattening-threshold: 1
+  output-folder: $(csharp-sdks-folder)/Subscriptions/Subscriptions/Generated
```
Comment (deathly809) [inline specification/azsadmin/resource-manager/user-subscriptions/readme.md]: please change to this,
output-folder: $(csharp-sdks-folder)/Generated

**Comment by deathly809** [comment]
@sarangan12 How does everything look?

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**